### PR TITLE
Update proxy sha and enable blackholecluster integration tests

### PR DESCRIPTION
**Pull request by jacob-delgado**
Last PR to fix https://github.com/istio/istio/issues/21385 and to enable telemetry v2 integration tests

### CODE REVIEW

File: tests/integration/mixer/outboundtrafficpolicy/v2/traffic_registry_only_test.go [hunk 1]
```diff
@@ -58,7 +56,7 @@ func TestOutboundTrafficPolicy_RegistryOnly_TelemetryV2(t *testing.T) {
 			Gateway:  true,
 			Expected: outboundtrafficpolicy.Expected{
 				Metric:          "istio_requests_total",
-				PromQueryFormat: `sum(istio_requests_total{destination_service_name="istio-egressgateway",response_code="200"})`,
+				PromQueryFormat: `sum(istio_requests_total{destination_service_name="istio-egressgateway.istio-system.svc.cluster.local",response_code="200"})`,
```
Comment (jacob-delgado) [inline tests/integration/mixer/outboundtrafficpolicy/v2/traffic_registry_only_test.go:59]: @nrjpoddar @mandarjog This was a change I noticed from mixer to telemetry v2 labels.

For mixer, the label created was `istio-egressgateway` and it's the fqdn in telemetry v2.

File: tests/integration/mixer/outboundtrafficpolicy/v2/traffic_registry_only_test.go [hunk 2]
```diff
@@ -67,17 +65,19 @@ func TestOutboundTrafficPolicy_RegistryOnly_TelemetryV2(t *testing.T) {
 			Name:     "TCP",
 			PortName: "tcp",
 			Expected: outboundtrafficpolicy.Expected{
-				Metric:          "istio_tcp_connections_closed_total",
-				PromQueryFormat: `sum(istio_tcp_connections_closed_total{reporter="source",destination_service_name="BlackHoleCluster",source_workload="client-v1"})`,
+				// TODO(https://github.com/istio/istio/issues/22717) re-enable TCP
```
Comment (gargnupur) [inline tests/integration/mixer/outboundtrafficpolicy/v2/traffic_registry_only_test.go:68]: From the logs in the bug, it does not look like they failed for Passthrough cluster and not Blackhole cluster?

Comment (jacob-delgado) [reply tests/integration/mixer/outboundtrafficpolicy/v2/traffic_registry_only_test.go:68]: John disabled them for mixer telemetry https://github.com/istio/istio/pull/22707/files#diff-9ca8ef00ed0f610d93a890f207e8dd96R66 due to flakiness. When I do my local testing I'm not able to replicate the flakiness. I can revert it and we can put up a PR if it is flaky.

Comment (gargnupur) [reply tests/integration/mixer/outboundtrafficpolicy/v2/traffic_registry_only_test.go:68]: Sounds good !

**Pull request merged**